drivers/interrupt_controller/intc_ioapic.c

```diff
@@ -438,7 +438,7 @@ static uint32_t __IoApicGet(int32_t offset)
 
 	key = irq_lock();
 
-	*((volatile uint32_t *) (IOAPIC_REG + IOAPIC_IND)) = (char)offset;
+	*((volatile uint32_t *) (IOAPIC_REG + IOAPIC_IND)) = offset;
```

Comment:
If any, it should be casting to `unsigned char` here since the upper 24 bits of the `IOREGSEL` register are reserved (must be 0).

Reply:
Thanks for the review, Got it fixed!